jena-arq/src/main/java/org/apache/jena/sparql/engine/http/QueryEngineHTTP.java

```diff
@@ -135,8 +135,7 @@ private QueryEngineHTTP(String serviceURI, Query query, String queryString, Http
         this.query = query;
         this.queryString = queryString;
         this.service = serviceURI;
-        // Copy the global context to freeze it.
-        this.context = new Context(ARQ.getContext());
+        this.context = ARQ.getContext();
```

Comment:
Yeah, I meant to ask about that. That (copying/freezing) the context was causing tests to fail in [TestAuth](https://github.com/apache/jena/blob/master/jena-fuseki1/src/test/java/org/apache/jena/fuseki/TestAuth.java#L204), because those tests involve possibly retrieving a client from the context inside `QueryEngineHTTP`. (The old form was to retrieve an `HttpAuthenticator` from the context.) Those tests first built the query object and then put the config into the context, so it wasn't available at execution, because I made the retrieval of config from the context happen as late as possible, at query execution. Would you suggest having it happen earlier, at construction time for `QueryEngineHTTP`?